core/src/main/java/org/apache/carbondata/core/scan/result/vector/impl/directread/ColumnarVectorWrapperDirectWithDeleteDeltaAndInvertedIndex.java

```diff
@@ -78,6 +80,17 @@ public void putNull(int rowId) {
     }
   }
 
+  @Override
+  public void putObject(int rowId, Object obj) {
+    if (!deletedRows.get(rowId)) {
+      if (nullBits.get(rowId)) {
+        columnVector.putNull(counter++);
```

Comment:
can you locally test one table with inverted index, I think here should not use `counter`, may be ned to use `rowId`, convert function is having `counter` already

Reply:
Have raised a separate jira to address this issue - CARBONDATA-4004